app/src/main/java/org/mozilla/focus/browser/integration/FullScreenIntegration.kt

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.browser.integration
+
+import android.app.Activity
+import android.os.Build
+import android.view.View
+import android.view.WindowManager
+import androidx.annotation.RequiresApi
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.EngineView
+import mozilla.components.feature.session.FullScreenFeature
+import mozilla.components.feature.session.SessionUseCases
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.base.feature.UserInteractionHandler
+import org.mozilla.focus.browser.DisplayToolbar
+
+class FullScreenIntegration(
+    val activity: Activity,
+    val sessionManager: SessionManager,
+    tabId: String?,
+    private val toolbarView: DisplayToolbar,
+    private val engineView: EngineView,
+    private val statusBar: View
+) : LifecycleAwareFeature, UserInteractionHandler {
+    private val feature = FullScreenFeature(
+        sessionManager,
+        SessionUseCases(sessionManager),
+        tabId,
+        ::viewportFitChanged,
+        ::fullScreenChanged
+    )
+
+    override fun start() {
+        feature.start()
+    }
+
+    override fun stop() {
+        feature.stop()
+    }
+
+    private fun fullScreenChanged(enabled: Boolean) {
+        if (enabled) {
+            toolbarView.visibility = View.GONE
```

Comment:
Setting the `toolbarView` to `GONE` shows a grey bar at the top of the screen and `setExpanded` no longer works.

<img width="360" alt="Screen Shot 2020-06-22 at 10 02 25 PM" src="https://user-images.githubusercontent.com/1370580/85352808-caff0b00-b4d4-11ea-9da4-5b301a17b43e.png">


I think we need to just call `setExpanded` to hide the toolbar, and then disable the toolbar behaviour when in fullscreen mode.

Not sure if you had a chance to take a look at this patch that does this: https://github.com/jonalmeida/focus-android/commit/6ef407097535917e73a22d64d29c08aad33632ed

Maybe there are other ways though?